lib/ae_mdw/db/sync/transaction.ex

```diff
@@ -223,6 +226,61 @@ defmodule AeMdw.Db.Sync.Transaction do
     {mutations, {txi + length(mb_txs), mbi + 1}}
   end
 
+  defp tx_mutations(
+         :contract_create_tx,
+         tx,
+         _signed_tx,
+         txi,
+         tx_hash,
+         _block_index,
+         block_hash,
+         mb_events
+       ) do
+    contract_pk = :aect_create_tx.contract_pubkey(tx)
+    owner_pk = :aect_create_tx.owner_pubkey(tx)
+    events = Map.get(mb_events, tx_hash, [])
+
+    mutations = [
+      ContractEventsMutation.new(contract_pk, events, txi)
+      | origin_mutations(:contract_create_tx, nil, contract_pk, txi, tx_hash)
+    ]
+
+    case Contract.get_info(contract_pk) do
+      {:ok, contract_info} ->
+        call_rec = Contract.get_init_call_rec(contract_pk, tx, block_hash)
+
+        aex9_meta_info =
+          if Contract.is_aex9?(contract_info) do
+            Contract.aex9_meta_info(contract_pk)
+          end
+
+        [
+          mutations,
```

Comment:
This is harmless because of the flatten happening outside but to decoupling might be better to append to the mutations (append to a small list of only 2 or 3 items).
I was going to propose putting it to the head of `mutations` list (like line 244) but `ContractCreateMutation.mutate` has a `DBContract.logs_write` call which depends on `Origin.tx_index` so better keep the same order as it's already done.

Reply:
Done